source/config/__init__.py

```diff
@@ -181,61 +245,145 @@ def initConfigPath(configPath=None):
+	Checks related easeOfAccess current user registry keys on Windows 8 or newer.
+	Then, checks the registry run key to see if NVDA
+	has been registered to start after logon on Windows 7
+	or by earlier NVDA versions.
+	"""
 	if (
 		easeOfAccess.canConfigTerminateOnDesktopSwitch
-		and easeOfAccess.willAutoStart(winreg.HKEY_CURRENT_USER)
+		and easeOfAccess.willAutoStart(easeOfAccess.AutoStartContext.AFTER_LOGON)
 	):
 		return True
 	try:
-		k = winreg.OpenKey(winreg.HKEY_CURRENT_USER, RUN_REGKEY)
-		val = winreg.QueryValueEx(k, u"nvda")[0]
-		return os.stat(val) == os.stat(sys.argv[0])
-	except (WindowsError, OSError):
+		k = winreg.OpenKey(winreg.HKEY_CURRENT_USER, RegistryKey.RUN.value)
+	except FileNotFoundError:
+		log.debugWarning(
+			f"Unable to find run registry key {RegistryKey.RUN}",
+			exc_info=True
+		)
+		return False
+	except WindowsError:
+		log.error(
+			f"Unable to open run registry key {RegistryKey.RUN}",
+			exc_info=True
+		)
 		return False
 
-def setStartAfterLogon(enable):
+	try:
+		val = winreg.QueryValueEx(k, "nvda")[0]
+	except FileNotFoundError:
+		log.debug("NVDA is not set to start after logon")
+		return False
+	except WindowsError:
+		log.error("Failed to query NVDA key to set start after logon", exc_info=True)
+		return False
+
+	try:
+		startAfterLogonPath = os.stat(val)
+	except WindowsError:
+		log.error(
+			"Failed to access the start after logon directory.",
+			exc_info=True
+		)
+		return False
+	
+	try:
+		currentSourcePath = os.stat(sys.argv[0])
+	except FileNotFoundError:
+		log.debug("Failed to access the current running NVDA directory.")
+		return False
+	except WindowsError:
+		log.error(
+			"Failed to access the current running NVDA directory.",
+			exc_info=True
+		)
+		return False
+
+	return currentSourcePath == startAfterLogonPath
+
+
+def setStartAfterLogon(enable: bool) -> None:
+	"""Not to be confused with setStartOnLogonScreen.
+	
+	Toggle if NVDA automatically starts after a logon.
+	Sets easeOfAccess related registry keys on Windows 8 or newer.
+
+	On Windows 7 this sets the registry run key.
+
+	When toggling off, always delete the registry run key to
+	in case it was set by an earlier version of NVDA or on Windows 7 or earlier.
+	"""
 	if getStartAfterLogon() == enable:
 		return
 	if easeOfAccess.canConfigTerminateOnDesktopSwitch:
-		easeOfAccess.setAutoStart(winreg.HKEY_CURRENT_USER, enable)
+		easeOfAccess.setAutoStart(easeOfAccess.AutoStartContext.AFTER_LOGON, enable)
 		if enable:
 			return
 		# We're disabling, so ensure the run key is cleared,
 		# as it might have been set by an old version.
 		run = False
 	else:
 		run = enable
-	k = winreg.OpenKey(winreg.HKEY_CURRENT_USER, RUN_REGKEY, 0, winreg.KEY_WRITE)
+	k = winreg.OpenKey(winreg.HKEY_CURRENT_USER, RegistryKey.RUN.value, 0, winreg.KEY_WRITE)
 	if run:
-		winreg.SetValueEx(k, u"nvda", None, winreg.REG_SZ, sys.argv[0])
+		winreg.SetValueEx(k, "nvda", None, winreg.REG_SZ, sys.argv[0])

```

Comment:
Could either of these fail?

Reply:
potentially, but I would expect an error to have been reported and logged as these exceptions aren't caught.
I was aiming to not change any behaviour in this PR.

Reply:
Ok, no problem.